update upsample tests in test_nn.py to test for memory_format

bdhirsh opened the pull request:
@ngimel pointed out to me where we already test the behavior of the `Upsample` ops in `test_nn.py`. This PR deleting my bespoke tests in `test_torch.py` and updates those in `test_nn.py` to test memory format properly.

There were two reasons the original test didn't pick up on a memory format regression:
- They didn't test the memory format of the output tensor explicitly, i.e. `output.is_contiguous(memory_format=...)`
- Even with that change, the test tensors were to simple to fail the tests. From some trial and error, it looks like one of the first two dimensions in the inputs needs to be > 1 in order for the `channels_last` memory format to actually re-order the strides.

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#53665 update upsample tests in test_nn.py to test for memory_format**
* #53535 fix channels_last bug in upsample kernels

Differential Revision: [D26929683](https://our.internmc.facebook.com/intern/diff/D26929683)

bdhirsh committed:
update upsample tests in test_nn.py to test for memory_format

[ghstack-poisoned]

bdhirsh commented on test/test_nn.py:
```diff
@@ -9046,17 +9048,21 @@ def test_upsamplingBilinear2d(self):
-                with warnings.catch_warnings(record=True) as w:
-                    out_t = m(in_t)
-                self.assertEqual(torch.ones(1, 1, out_size, out_size), out_t.data)
+            for memory_format in [torch.contiguous_format, torch.channels_last]:
```
GitHub displays this with a lot of red and green, but all I really did was add a for loop here to loop over the different memory formats. Then further down I assert that the output is contiguous w.r.t the expected memory format.

bdhirsh committed:
Update on "update upsample tests in test_nn.py to test for memory_format"

@ngimel pointed out to me where we already test the behavior of the `Upsample` ops in `test_nn.py`. This PR deleting my bespoke tests in `test_torch.py` and updates those in `test_nn.py` to test memory format properly.

There were two reasons the original test didn't pick up on a memory format regression:
- They didn't test the memory format of the output tensor explicitly, i.e. `output.is_contiguous(memory_format=...)`
- Even with that change, the test tensors were to simple to fail the tests. From some trial and error, it looks like one of the first two dimensions in the inputs needs to be > 1 in order for the `channels_last` memory format to actually re-order the strides.


Differential Revision: [D26929683](https://our.internmc.facebook.com/intern/diff/D26929683)

[ghstack-poisoned]

ngimel commented on test/test_nn.py:
```diff
@@ -9015,27 +9015,29 @@ def test_upsamplingLinear1d_spatial_invariance(self):
                 out_uint8_t = m(in_uint8_t)
-            self.assertEqual(torch.ones(1, 1, 4, 4).contiguous(memory_format=memory_format), out_t.data)
-            self.assertEqual(torch.ones(1, 1, 4, 4, dtype=torch.uint8).contiguous(memory_format=memory_format), out_uint8_t.data)
+            self.assertEqual(torch.ones(1, 2, 4, 4).contiguous(memory_format=memory_format), out_t.data)
```
nit: `.data` is deprecated API, and you don't need to call `.contiguous()` on the expected tensor, so 
```
self.assertEqual(torch.ones(...), out_t)
```

ngimel commented on test/test_nn.py:
```diff
@@ -9015,27 +9015,29 @@ def test_upsamplingLinear1d_spatial_invariance(self):
-            self.assertEqual(torch.ones(1, 1, 4, 4).contiguous(memory_format=memory_format), out_t.data)
-            self.assertEqual(torch.ones(1, 1, 4, 4, dtype=torch.uint8).contiguous(memory_format=memory_format), out_uint8_t.data)
+            self.assertEqual(torch.ones(1, 2, 4, 4).contiguous(memory_format=memory_format), out_t.data)
+            self.assertEqual(torch.ones(1, 2, 4, 4, dtype=torch.uint8).contiguous(memory_format=memory_format), out_uint8_t.data)
```
same comment here

ngimel commented on test/test_nn.py:
```diff
@@ -9015,27 +9015,29 @@ def test_upsamplingLinear1d_spatial_invariance(self):
+            self.assertEqual(torch.ones(1, 2, 4, 4).contiguous(memory_format=memory_format), out_t.data)
+            self.assertEqual(torch.ones(1, 2, 4, 4, dtype=torch.uint8).contiguous(memory_format=memory_format), out_uint8_t.data)
+            # Assert that memory format is carried through to the output
+            self.assertTrue(out_t.is_contiguous(memory_format=memory_format))
```
That's good!

ngimel commented on test/test_nn.py:
```diff
@@ -9015,27 +9015,29 @@ def test_upsamplingLinear1d_spatial_invariance(self):
             with warnings.catch_warnings(record=True) as w:
                 out_t = m(in_t)
-            self.assertEqual(torch.ones(1, 1, 4, 2).contiguous(memory_format=memory_format), out_t.data)
+            self.assertEqual(torch.ones(1, 2, 4, 2).contiguous(memory_format=memory_format), out_t.data)
```
probably makes sense to test memory_format here too?

ngimel reviewed (approved):
Thanks! Please get rid of .data attributes, otherwise looks good.

bdhirsh committed:
Update on "update upsample tests in test_nn.py to test for memory_format"

@ngimel pointed out to me where we already test the behavior of the `Upsample` ops in `test_nn.py`. This PR deleting my bespoke tests in `test_torch.py` and updates those in `test_nn.py` to test memory format properly.

There were two reasons the original test didn't pick up on a memory format regression:
- They didn't test the memory format of the output tensor explicitly, i.e. `output.is_contiguous(memory_format=...)`
- Even with that change, the test tensors were to simple to fail the tests. From some trial and error, it looks like one of the first two dimensions in the inputs needs to be > 1 in order for the `channels_last` memory format to actually re-order the strides.


Differential Revision: [D26929683](https://our.internmc.facebook.com/intern/diff/D26929683)

[ghstack-poisoned]

ezyang commented:
Thanks a lot for following up on this!
